spec/slack_markdown/processor_spec.rb

```diff
@@ -23,14 +23,15 @@
 
   let :text do
     <<~EOS
-      <@U12345> <@U23456> <#C01S1JQMYKV|ru_shalm> *SlackMarkdown* is `text formatter` ~package~ _gem_ .
+      <@U12345> <@U23456> <#C01S1JQMYKV> *SlackMarkdown* is `text formatter` ~package~ _gem_ .
       > :ru_shalm: is <http://toripota.com/img/ru_shalm.png>
+      <@U12345|dont_override_me> <@U23456|override_me> <#C01S1JQMYKV|dont_override_me> <#C12345|override_me>
     EOS
   end
 
   it do
     should eq "<a href=\"/@ru_shalm\" class=\"mention\">@ru_shalm</a> @U23456 <a href=\"http://localhost/?url=http%3A%2F%2Ftoripota.com\" class=\"channel\">#ru_shalm</a> <b>SlackMarkdown</b> is <code>text formatter</code> <strike>package</strike> <i>gem</i> .<br><blockquote>
 <img class=\"emoji\" title=\":ru_shalm:\" alt=\":ru_shalm:\" src=\"http://toripota.com/img/ru_shalm.png\" height=\"20\" width=\"20\" align=\"absmiddle\"> is <a href=\"http://localhost/?url=http%3A%2F%2Ftoripota.com%2Fimg%2Fru_shalm.png\" class=\"link\">http://toripota.com/img/ru_shalm.png</a><br>
-</blockquote>"
+</blockquote><a href=\"/@ru_shalm\" class=\"mention\">@ru_shalm</a> override_me <a href=\"http://localhost/?url=http%3A%2F%2Ftoripota.com\" class=\"channel\">#ru_shalm</a> <a href=\"#C12345\" class=\"channel\">override_me</a><br>"
```

Comment:
Side-note; I was a little surprised by the result in this test, which is that if the `on_slack_user_id` block returns `nil` for a user, we just render the override text and no link… But for channels, we always render a link, even if the `on_slack_channel_id` block returns `nil`. Maybe this is intentional for some reason? But if there's a consistent behavior you'd rather see, I can fix that up in this pull request too

Reply:
ummm... I wrote this code a long time ago, I don't remember it very accurately. :sweat:
Probably I thought it was a good idea to display the information as it was, in case we could not retrieve it for some reason.

(ex.private channel link)
<img width="178" alt="20211101_230613_chrome_apGXk3ACJe" src="https://user-images.githubusercontent.com/1131422/139685747-815e45fe-5b21-42bc-91d8-212baebb6978.png">